### PR TITLE
feat: add admin docs for CVMs

### PR DESCRIPTION
**Pull request by cmoog**
This will replace the existing CVM article in the help center. We can follow-up this PR with cloud-specific deployment docs in the help center and link to them from here.

### CODE REVIEW

File: admin/environment-management/cvms.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+deployment option. This option allows users to run system-level
+programs inside their Environment including Docker and systemd.
+
+Navigate to **Manage > Admin > General** to enable this option.
+
+## Infrastructure requirements
+
+- The Kubernetes Nodes must have a minimum kernel version of `5.4`
+(released Nov 24th, 2019).
+- The Kubernetes Nodes must be running an Ubunutu OS.
```
Comment (nathanpotter) [inline admin/environment-management/cvms.md]:
```suggestion
- The Kubernetes Nodes must be running an Ubuntu OS.
```

Comment (nathanpotter) [reply admin/environment-management/cvms.md]: Is ubuntu the only supported option for sysbox?

File: admin/environment-management/cvms.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+- Legacy versions of cluster-wide proxy services such as Istio are not
```
Comment (nathanpotter) [inline admin/environment-management/cvms.md:18]: Do you know which versions specifically?

File: admin/environment-management/cvms.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+deployment option. This option allows users to run system-level
+programs inside their Environment including Docker and systemd.
+
+Navigate to **Manage > Admin > General** to enable this option.
+
+## Infrastructure requirements
+
+- The Kubernetes Nodes must have a minimum kernel version of `5.4`
+(released Nov 24th, 2019).
+- The Kubernetes Nodes must be running an Ubuntu OS.
```
Comment (nathanpotter) [inline admin/environment-management/cvms.md:17]: Is this a sysbox restriction or are we trying to limit the OSes we need to support?

File: admin/environment-management/cvms.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+lifecycle of their Environment. The isolation between the user’s Environment
+container and it’s outer, supervising container is what provides
+[strong isolation](https://github.com/nestybox/sysbox/blob/master/docs/user-guide/security.md).
+
+## Image Configuration
+
+### systemd
+
+During Environment startup, Coder checks for the presence of `/sbin/init` within
+the Environment Image. If the file exists, it's used as the container entrypoint
```
Comment (nathanpotter) [inline admin/environment-management/cvms.md:44]: I think we should specify what happens if `/sbin/init` doesn't exist, which I believe is we just start our own init process – tinyinit or something.

File: admin/environment-management/cvms.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+RUN ln -s /lib/systemd/systemd /sbin/init
+```
+
+### Adding Docker
+
+Be sure to install the `docker` packages into your Environment Image. For a
+seamless experience, use [systemd](#systemd) and register the `docker` service
+so `dockerd` is automatically run during initialization.
+
+The following snippet demonstrates how an image can registry the `docker`
```
Comment (nathanpotter) [inline admin/environment-management/cvms.md:67]:
```suggestion
The following snippet demonstrates how an image can register the `docker`
```

File: admin/environment-management/cvms.md (hunk 3 above)
Comment (cmoog) [reply admin/environment-management/cvms.md:17]: @kylecarbs doesn't think this is *needed* but if I recall correctly the sysbox maintainers suggested it as a requirement. So including it in our docs feels like the safe bet either way.

Comment (kylecarbs) [reply admin/environment-management/cvms.md:17]: So with the `5.4` restriction I don't believe we need to limit to Ubuntu distributions... I can say though that this is all we've tested on, so might be risky to do anything else.

File: admin/environment-management/cvms.md (hunk 2 above)
Comment (kylecarbs) [reply admin/environment-management/cvms.md:18]: Istio <= 1.14 is supported. @deansheather has better intel on these verisons.

Comment (cmoog) [reply admin/environment-management/cvms.md:18]: The phrasing here is intentionally vague until we have additional confidence for the combinations of Istio version and Kube version that work properly. Basically, not exactly

File: admin/environment-management/cvms.md (hunk 3 above)
Comment (nathanpotter) [reply admin/environment-management/cvms.md:17]: agree, let's specify it and someone can ask for another distro if they want it – at that point we can test it out if it makes sense

File: admin/environment-management/cvms.md (hunk 4 above)
Comment (cmoog) [reply admin/environment-management/cvms.md:44]: So technically we use `sleep infinity` as the entrypoint if no `/sbin/init` is found. This ensures that the Coder init is always executed as a container exec in both code paths...

Comment (cmoog) [reply admin/environment-management/cvms.md:44]: Either way, the exact behavior in the case of no `/sbin/init` feels like an implementation detail.

File: admin/environment-management/cvms.md (hunk 2 above)
Comment (deansheather) [reply admin/environment-management/cvms.md:18]: Istio 1.5 or lower

Or newest Istio and k8s 1.19

Comment (deansheather) [reply admin/environment-management/cvms.md:18]: Actually I meant 1.4 or lower. My bad

Comment (deansheather) [reply admin/environment-management/cvms.md:18]: Actually it's 1.4 or lower

Comment (cmoog) [reply admin/environment-management/cvms.md:18]: Let's keep it vague until we're more confident from our own testing or until we have evidence from more customers that it's compatible with their cluster configurations.

**Review by kylecarbs — APPROVED** [review]

**Pull request merged**